src/twisted/internet/process.py

```diff
@@ -129,6 +133,34 @@ def detectLinuxBrokenPipeBehavior():
 
 
 
+def detectCygwinBrokenPipeBehavior():
+    """
+    On Cygwin, polling on the write end of a pipe whose read end has been
+    closed does not set the POLLERR bit.
+
+    In this case we should not use the "enableReadHack" flag of
+    L{ProcessWriter.doRead} or else a select()/poll() on that file descriptor
+    will block.
+
+    @return: C{True} if Cygwin pipe behaviour is broken.
+    @rtype : L{bool}
+    """
+
+    if sys.platform != 'cygwin':
+        return False
+
+    r, w = os.pipe()
+    os.close(r)
+    reads, writes, excs = select.select([w], [], [], 0)
+    os.close(w)
+    return not reads
```

Comment:
I don't quite understand why we get this from here.

From the Trac ticket, my understand is that Cygwin always has this behaviour as there versions of Cygwin in which the pipes are not "broken"?

The docstring on this method does not contain "On some Cygwin" versions